pkg/sdk/application.go

```diff
@@ -0,0 +1,37 @@
+package sdk
+
+import (
+	"context"
+
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+)
+
+// ApplicationSuppliers is the structure that represents a session's end block height
+// and its matching suppliers.
+type ApplicationSuppliers struct {
+	// Application is the fully hydrated session object returned by the query.
+	Application []apptypes.Application
+}
+
+// GetApplicationSupplierEndpoints returns a flattened structure of the endpoints
+// from all suppliers in the session and returns them as a SupplierEndpoint slice.
+// It queries for the latest session and caches it if the cached one is outdated.
+func (sdk *poktrollSDK) GetApplications(ctx context.Context, gateway string) ([]apptypes.Application, error) {
+
+	filtered := []apptypes.Application{}
```

Comment:
Could you replace the `filtered` variable name by a more contextual one? I would suggest `delegatingApplications`

cc @Olshansk , a better suggestion?

Reply:
Done.